testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_filtering.py

```diff
@@ -0,0 +1,585 @@
+        true.
+        4. Verify SG has new fields added only when "new_field_1" is true
+    """
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    if sync_gateway_version < "2.5.0":
+        pytest.skip('This test cannnot run with sg version below 2.5')
+    channels = ["ABC"]
+    username = "autotest"
+    password = "password"
+
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # 1. Creating docs in CBL app
+    sg_client = MobileRestClient()
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name=username, password=password, channels=channels)
+    cookie, session = sg_client.create_session(url=sg_admin_url, db=sg_db, name=username)
+    sync_cookie = "{}={}".format(cookie, session)
+    session_header = {"Cookie": sync_cookie}
+    auth_session = cookie, session
+    db.create_bulk_docs(num_of_docs, "cbl", db=cbl_db, channels=channels)
+
+    # 2. Replicating docs to SG
+    # Configure replication with push/pull
+    replicator = Replication(base_url)
+    repl_config = replicator.configure(source_db=cbl_db,
+                                       target_url=sg_blip_url,
+                                       continuous=False,
+                                       headers=session_header)
+    repl = replicator.create(repl_config)
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    assert total == completed, "total is not equal to completed"
+    replicator.stop(repl)
+
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, include_docs=True)["rows"]
+
+    # Verify database doc counts
+    cbl_doc_count = db.getCount(cbl_db)
+    assert len(sg_docs) == cbl_doc_count, "Expected number of docs does not exist in sync-gateway after replication"
+
+    # 3. Modify docs in SG so that we can do pull replication. The replication will have filter for newly added field
+    # Docs with new_field_1 value to true will only be replicated, others will be rejected by filter method
+    for item in sg_docs:
+        sg_doc = item["doc"]
+        sg_client.update_doc(url=sg_url, db=sg_db, doc_id=sg_doc["_id"],
+                             number_updates=1, auth=auth_session,
+                             property_updater=add_new_fields_to_doc)
+    repl_config = replicator.configure(source_db=cbl_db,
```

Comment:
From design, debugging and readability perspective it's better to use it this way, as it replicate and configure does all this thing in one go and in case of failure we have to trace back as it in not written with SRP principle.